src/backend/access/appendonly/aosegfiles.c

```diff
@@ -1496,6 +1502,7 @@ get_ao_distribution(PG_FUNCTION_ARGS)
 		SRF_RETURN_NEXT(funcctx, result);
 	}
 
+srf_done:
 	/*
 	 * do when there is no more left
 	 */
```

Comment:
I think, we need some check here, see https://github.com/arenadata/gpdb/pull/753

Reply:
I agree, but that check seems out of scope for this issue.

Reply:
I don't think so. You added a new path with the `goto` statement, in which the `query_block` pointer may be NULL and/or the `SPI_connect` function has not yet been called.

Reply:
It seems safe to call SPI_finish without active connection. SPI_finish has a protection for this case:

```
	res = _SPI_begin_call(false);	/* just check we're connected */
	if (res < 0)
		return res;

```